components/service/digital-asset-links/src/main/java/mozilla/components/service/digitalassetlinks/local/StatementRelationChecker.kt

```diff
@@ -0,0 +1,32 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.digitalassetlinks.local
+
+import mozilla.components.service.digitalassetlinks.AssetDescriptor
+import mozilla.components.service.digitalassetlinks.Relation
+import mozilla.components.service.digitalassetlinks.RelationChecker
+import mozilla.components.service.digitalassetlinks.Statement
+import mozilla.components.service.digitalassetlinks.StatementLister
+
+class StatementRelationChecker(
```

Comment:
For the DAL impl, we kept the `RelationChecker` and `StatementLister` together but this depends on each other so we separated them into two? I like that their separate, but I was curious about the reasoning behind the DAL.

Reply:
For custom tabs, we only care about the "check" step. The "list" step is a separate API call that we haven't used. However, to do checking locally, we need to iterate over the list of avaliable options.

The API-based DAL class supports both in a single implementation because most of the resource fetching code is shared. The local versions are split because local checking must have a list to iterate through.